examples/http-cache-middleware/index.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+use Dumbo\Dumbo;
+use Dumbo\Middleware\CacheMiddleware;
+
+require "vendor/autoload.php";
+
+
+$app = new Dumbo();
+
+$app->use(CacheMiddleware::withHeaders(
```

Comment:
Does it make sense to show applying different cache rules for a group of sub routes? I'm actually not sure what we have now would support it, but we should figure it out (maybe not now?). Let me know what you think.

Reply:
I don't know if I've understood the question correctly but I think that it makes total sense, I was thinking of something like that to apply cache rules for a particular route or group of routes  : 

```php
$app->get('/hello', function () {

})
->cache([
...
])
```
Let me know if this is what you meant ?

Reply:
Yes, it does make sense to me! What would be the best way to do it?

I see two ways of doing this :

Chaining the methods together 

```php
$app
    ->cache([
        "type" => "private",
        "maxAge" => 3600,
        "mustRevalidate" => true,
        "strictEtag" => true,
    ])
    ->get("/hello", function ($c) {
...
    });
```

Or by adding a url parameter to the middleware that would allow it to be applied to every route that matches

```php
$app->use('/api/*', CacheMiddleware::withHeaders(
    type: "public",
    mustRevalidate: true,
    maxAge: 3600,
    strictEtag: true
));
```

Reply:
Hey @AymaneIsmail, the second example you provided would fit very nicely into the updates I pushed earlier. See an example here for setting some context for the `/api` route only: https://github.com/notrab/dumbo/blob/main/examples/context-variables/index.php#L25

Reply:
There might be an addiitonal change required so it works on nested routes, I haven't tried that yet.